src/EnergyPlus/Plant/PlantLocation.hh

```diff
@@ -52,14 +52,28 @@
 
 namespace EnergyPlus {
 
+  
+// Forward declarations
+namespace DataPlant {
+    struct PlantLoopData;
+    struct HalfLoopData;
+    struct BranchData;
+    struct CompData;
+}
+
 struct PlantLocation
 {
     // Members
-    int loopNum;
+    int loopNum = -1;
     DataPlant::LoopSideLocation loopSideNum;
-    int branchNum;
-    int compNum;
+    int branchNum = -1;
+    int compNum = -1;
 
+    DataPlant::PlantLoopData *loop = nullptr;
```

Comment:
New fields.

Reply:
Does the constructor below overwrite these initializations?

PlantLocation() : loopNum(0), loopSideNum(DataPlant::LoopSideLocation::Invalid), branchNum(0), compNum(0)
    {
    }

Reply:
That's a very good question. I didn't know the answer to this so I looked at it in the deubbger and the explicit default constructor does overwrite the default values given during the declaration. That's not good. C++ should not allow you to have both for the same variable. I believe we are moving away from explicit default constructors, so I will delete this one and test that -1's work just as well, otherwise I will change them to 0. Good catch.

Reply:
Ok, it turns out that if you delete the explicit default constructor of class A, then you have to also delete the explicit default constructor of class B if class B embeds A. Given that I am not going to refactor the default constructors of all plant objects in this PR, I'm going to leave this as is and maybe come back to it later (FLW).